open_instruct/rejection_sampling/generation.py

```diff
@@ -146,22 +146,20 @@ def format_conversation(messages: list) -> str:
 
 def main(args: Args, dataset_args: DatasetArgs, gen_args: GenerationArgs):
 
-    ds = load_dataset(dataset_args.dataset_name)
+    ds = load_dataset(dataset_args.dataset_name, split=dataset_args.split)
     if dataset_args.sanity_check:
-        for key in ds:
-            ds[key] = ds[key].select(range(min(dataset_args.sanity_check_size, len(ds[key]))))
+        ds = ds.select(range(min(dataset_args.sanity_check_size, len(ds))))
     if dataset_args.dataset_end_idx is None:
-        dataset_args.dataset_end_idx = len(ds[dataset_args.dataset_train_split])
-    for key in ds:
-        ds[key] = ds[key].select(range(dataset_args.dataset_start_idx, dataset_args.dataset_end_idx))
+        dataset_args.dataset_end_idx = len(ds)
+    ds = ds.select(range(dataset_args.dataset_start_idx, dataset_args.dataset_end_idx))
```

Comment:
We are only using `ds[dataset_args.dataset_train_split]["prompt"]` anyway. Might as well just select the proper split